packages/docusaurus/src/commands/deploy.ts

```diff
@@ -184,33 +185,31 @@ You can also set the deploymentBranch property in docusaurus.config.js .`);
   // out to deployment branch.
   const currentCommit = shellExecLog('git rev-parse HEAD').stdout.trim();
 
-  const runDeploy = async (outputDirectory: string) => {
+  const runDeploy = async (
+    outputDirectory: string,
+    targetDirectory: string,
+  ) => {
     const fromPath = outputDirectory;
     const toPath = await fs.mkdtemp(
       path.join(os.tmpdir(), `${projectName}-${deploymentBranch}`),
     );
     shell.cd(toPath);
 
-    // Check out deployment branch when cloning repository, and then remove all
-    // the files in the directory. If the 'clone' command fails, assume that
-    // the deployment branch doesn't exist, and initialize git in an empty
-    // directory, check out a clean deployment branch and add remote.
-    if (
-      shellExecLog(
-        `git clone --depth 1 --branch ${deploymentBranch} ${deploymentRepoURL} "${toPath}"`,
-      ).code === 0
-    ) {
-      shellExecLog('git rm -rf .');
-    } else {
-      shellExecLog('git init');
-      shellExecLog(`git checkout -b ${deploymentBranch}`);
-      shellExecLog(`git remote add origin ${deploymentRepoURL}`);
-    }
+    // Clones the deployment repository, then checks out the deployment
+    // branch and removes all files in the target directory.
+    // If the branch doesn't exist, it is created.
+    // This preserves the existing files outside of the target directory
+    // from being modified.
+    shellExecLog(`git clone --depth 1 ${deploymentRepoURL} .`);
+    shellExecLog(`git checkout -B ${deploymentBranch}`);
+    shellExecLog(`git remote add origin ${deploymentRepoURL}`);
```

Comment:
Chore: need to correct these lines here, this will always create a new branch since we fetch with depth 1 on the line above. Need to fetch information about any existing branch before calling checkout with `-B`.